projects/igniteui-angular/src/lib/grids/pivot-grid/pivot-grid.interface.ts

```diff
@@ -57,3 +58,78 @@ export enum PivotDimensionType {
     Row,
     Column
 }
+
+// Equals to pretty much this configuration:
+// {
+//     member: () => 'All Periods',
+//     enabled: true,
+//     fieldName: 'AllPeriods',
+//     childLevel: {
+//         fieldName: 'Years',
+//         member: (rec) => {
+//             const recordValue = rec['Date'];
+//             return recordValue ? (new Date(recordValue)).getFullYear().toString() : rec['Years'];
+//         },
+//         enabled: true,
+//         childLevel: {
+//                 member: (rec) => {
+//                     const recordValue = rec['Date'];
+//                     return recordValue ? new Date(recordValue).toLocaleString('default', { month: 'long' }) : rec['Months'];
+//                 },
+//                 enabled: true,
+//                 fieldName: 'Months',
+//                 childLevel: {
+//                         member: 'Date',
+//                         fieldName:'Date',
+//                         enabled: true
+//                     }
+//             }
+//     }
+// },
+
+export class IgxPivotDateDimension implements IPivotDimension {
+    public childLevel: IPivotDimension;
+    public memberName = 'AllPeriods';
```

Comment:
If we include 'AllPeriods' it needs to be part of of the localized strings

Reply:
If it's something to be localized, it should be human-readable like 'All Periods'

Reply:
This is not the text that needs localizing, its the value that `memberFunction` returns that needs localizing.